webhooks/framework.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/rancher/webhook-service/drivers"
+)
+
+// ConstructPayload constructs jwt url
+func ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	if _, ok := input["driver"].(string); !ok {
+		return nil, fmt.Errorf("Driver of type string not provided\n")
+	}
+	driverId := input["driver"].(string)
+	driver := drivers.GetDriver(driverId)
+	if driver == nil {
+		return nil, fmt.Errorf("Driver %s is not registered\n", driverId)
+	}
+	payload, err := driver.ConstructPayload(input)
```

Comment:
We discussed making the ConstructPayload function generic, not driver specific. Why did you decide to add it back to the driver? You may have told me already, but I don't recall.